### DISCUSSION TITLE
Add Pages progmatically

### DISCUSSION THREAD

**Discussion by LahmerIlyas** [vercel/next.js#18733] [Help]
Hi, I'm using nextjs with a custom express server. 
I'm looking for a way to add pages programmatically rather than creating pages folder and putting all pages there.
Is this feature available, or can anybody shows me a way to do it without the need of pages folder.

**Comment by LahmerIlyas** [comment]
Anyone can answer me !

**Reply by balazsorban44** [reply]
Could you please explain your usecase? Would any of this help?: 

https://nextjs.org/docs/basic-features/data-fetching

https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes

**Reply by LahmerIlyas** [reply]
@balazsorban44 I want to embed nextjs in a package that will be a utility for developers to auto generate dashboards. So I can't rely on the pages directory.

**Reply by inonjs** [reply]
> @balazsorban44 I want to embed nextjs in a package that will be a utility for developers to auto generate dashboards. So I can't rely on the pages directory.

Me too! :/

**Reply by LahmerIlyas** [reply]
@balazsorban44 Do you have any idea how can we achieve this?

**Reply by balazsorban44** [reply]
Hm, unfortunately not. Maybe generating page files with Node, not Next.js itself

**Reply by inonjs** [reply]
> @balazsorban44 Do you have any idea how can we achieve this?

I'm trying with this way, but having issue with typescript transpilation. 

Repo: https://github.com/inonjs/next-typescript-with-embeded-pages
